### ISSUE TITLE
User assignment query should not show appeals that have been re-assigned to an attorney twice. 

### ISSUE THREAD

**Issue by amprokop** [department-of-veterans-affairs/caseflow#4476]
I'll just quote an email exchange with Jed for full context:

```
1. Caseflow Reader currently shows attorneys/judges a list of appeals that are assigned to them, but we've heard a few complaints that lead us to believe there is a bug in our query. If an appeal is assigned to an attorney, then it gets assigned back to a judge to get an outside medical opinion, then it is assigned back to the same attorney, apparently it is showing up twice in the list.

Here is the query code. Looks like this is because we do a LEFT JOIN on the DECASS table, which will give us one result row for each row in the DECASS table that matches. I believe that DECASS.DERECEIVE should be set to a date when the appeal returns to a judge, and so it should be null beforehand — so if we only return results WHERE DECASS.DERECEIVE is null, we will solve the problem. It looks like VACOLS does this check too — can you confirm that it's the correct approach?

 

When a case goes out for a VHA opinion the attorney ends up with two work products the VHA request and the final decision thus two entries in the DECASS table. This will create two rows in the queue when you join BRIEFF and DECASS.  I would use  ‘where DECOMP date is null’ to avoid this situation.  Using DERECEIVE would cause problems because sometime an attorney sends it to the Judge DERECEIVE gets updated but the Judge sends it back to the attorney and it charged to them again, you want it to show on their queue but the DERECEIVE will no longer be null.  DECOMP would be more reliable for filtering out duplicate DECASS entries. 


You also want to keep the left (outer) join and retrieve any records from BRIEFF where there is no DECASS record.
```

TL;DR — we need to add `‘where DECOMP is null’` to both the Reader query `select_assignments` (we'll remove this query after Queue launches) and to the Queue query (`select_tasks`).

Dev should connect with @laurjpeterson to verify that this ticket is complete with users. @laurjpeterson will be the final validator.

# AC
- When an appeal has been re-assigned to an attorney, it shouldn't appear twice in the user's list

**Comment by aroltsch** [comment]
I tested `bvajfreeman` in production and it is showing that he has 0 cases now. Asking @laurjpeterson  to validate.

```
result = VACOLS::CaseAssignment.tasks_for_user("bvajfreeman")
=> #<ActiveRecord::Relation []>
```

**Comment by aroltsch** [comment]
Josh said it is fixed.
